src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs

```diff
@@ -14,48 +15,88 @@
 using Microsoft.Health.Fhir.Core.Extensions;
+using Microsoft.Health.Fhir.Core.Features.Persistence;
 using Microsoft.Health.Fhir.Core.Messages.Operation;
+using Microsoft.Health.Fhir.Core.Models;
 using Microsoft.Health.Fhir.ValueSets;
 
 namespace Microsoft.Health.Fhir.Api.Controllers
 {
     [ServiceFilter(typeof(AuditLoggingFilterAttribute))]
     [ServiceFilter(typeof(OperationOutcomeExceptionFilterAttribute))]
     [ServiceFilter(typeof(ValidateContentTypeFilterAttribute))]
-    [ServiceFilter(typeof(ValidationQueryFilterAndParameterParserAttribute))]
     [ValidateResourceTypeFilter(true)]
     [ValidateModelState]
     public class ValidateController : Controller
     {
         private readonly IMediator _mediator;
+        private readonly ResourceDeserializer _resourceDeserializer;
 
-        public ValidateController(IMediator mediator)
+        public ValidateController(IMediator mediator, ResourceDeserializer resourceDeserializer)
         {
             EnsureArg.IsNotNull(mediator, nameof(mediator));
 
             _mediator = mediator;
+            _resourceDeserializer = resourceDeserializer;
         }
 
         [HttpPost]
         [Route(KnownRoutes.ValidateResourceType)]
         [AuditEventType(AuditEventSubType.Read)]
-        public async Task<IActionResult> Validate([FromBody] Resource resource)
+        public async Task<IActionResult> Validate([FromBody] Resource resource, [FromQuery(Name = "profile")] string profile)
         {
-            return await RunValidationAsync(resource);
+            Uri profileUri = null;
+            if (resource.ResourceType == ResourceType.Parameters)
+            {
+                var parameterResource = (Parameters)resource;
+                var profileFromParameters = parameterResource.Parameter.Find(param => param.Name.Equals("profile", StringComparison.OrdinalIgnoreCase));
+                if (profileFromParameters != null)
+                {
+                    if (profile != null)
+                    {
+                        throw new BadRequestException(Api.Resources.MultipleProfilesProvided);
+                    }
+
+                    if (profileFromParameters.Value != null)
+                    {
+                        profile = profileFromParameters.Value.ToString();
+                    }
+                }
+
+                resource = parameterResource.Parameter.Find(param => param.Name.Equals("resource", StringComparison.OrdinalIgnoreCase)).Resource;
+            }
+
+            if (!string.IsNullOrEmpty(profile))
+            {
+                try
+                {
+                    profileUri = new Uri(profile);
+                }
+                catch
+                {
+                    throw new BadRequestException(string.Format(Resources.ProfileIsInvalid, profile));
+                }
+            }
+
+            return await RunValidationAsync(resource.ToResourceElement(), profileUri);
         }
 
-        [HttpPost]
+        [HttpGet]
```

Comment:
Is there a reason this was changed to a Get whereas the above is a Post? The spec seems to indicate both are Post requests ([https://www.hl7.org/fhir/resource-operation-validate.html](https://www.hl7.org/fhir/resource-operation-validate.html))

Reply:
I return back http post

---
In reply to: [581635995](https://github.com/microsoft/fhir-server/pull/1654#discussion_r581635995) [](ancestors = 581635995)